src/components/fields/date-range-field/date-range-field.tsx

```diff
@@ -0,0 +1,259 @@
+
+const DEFAULT_DATE_FORMAT = "uuuu-MM-dd";
+const DEFAULT_DATE_FORMATTER = DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT)
+	.withResolverStyle(ResolverStyle.STRICT)
+	.withLocale(Locale.ENGLISH);
+
+export const DateRangeField = (props: IGenericFieldProps<IDateRangeFieldSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REF
+	// =============================================================================
+	const {
+		schema: { label, dateFormat = DEFAULT_DATE_FORMAT, validation, variant, ...otherSchema },
+		id,
+		isDirty,
+		onChange,
+		value = { from: undefined, to: undefined },
+		error,
+		...otherProps
+	} = props;
+	const [stateValue, setStateValue] = useState<string>(value.from || ""); // always uuuu-MM-dd because it is passed to Form.DateInput
+	const [stateValueEnd, setStateValueEnd] = useState<string>(value.to || ""); // always uuuu-MM-dd because it is passed to Form.DateInput
+	const [derivedProps, setDerivedProps] = useState<DateInputProps>();
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		const futureRule = validation?.find((rule) => "future" in rule);
+		const pastRule = validation?.find((rule) => "past" in rule);
+		const notFutureRule = validation?.find((rule) => "notFuture" in rule);
+		const notPastRule = validation?.find((rule) => "notPast" in rule);
+		const minDateRule = validation?.find((rule) => "minDate" in rule);
+		const maxDateRule = validation?.find((rule) => "maxDate" in rule);
+		const isRequiredRule = validation?.find((rule) => "required" in rule);
+		const excludedDatesRule = validation?.find((rule) => "excludedDates" in rule);
+
+		const minDate = DateTimeHelper.toLocalDateOrTime(minDateRule?.["minDate"], dateFormat, "date");
+		const maxDate = DateTimeHelper.toLocalDateOrTime(maxDateRule?.["maxDate"], dateFormat, "date");
+
+		setFieldValidationConfig(
+			id,
+			Yup.object()
+				.shape({
+					from: Yup.string(),
+					to: Yup.string(),
+				})
+				.test(
+					"is-empty-string",
+					isRequiredRule?.["errorMessage"] || ERROR_MESSAGES.DATE_RANGE.REQUIRED,
+					(value) => {
+						if (!value || !isRequiredRule) return true;
+						return value.from?.length > 0 && value.to?.length > 0;
+					}
+				)
+				.test("is-date", ERROR_MESSAGES.DATE_RANGE.INVALID, (value) => {
+					if (!value || value.from === "" || value.to === "") return true;
+					if (value.from === undefined || value.to === undefined) return true;
+					if (!isValidDate(value.from) || !isValidDate(value.to)) return false;
+					return (
+						!!DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date") ||
+						!!DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date")
+					);
```

Comment:
this will return truthy if either value.from or value.to is valid but both needs to be valid

Reply:
Fixed